test/beast/http/message_generator.cpp

```diff
@@ -0,0 +1,315 @@
+//
+// Copyright (c) 2022 Seth Heeren (sgheeren at gmail dot com)
+//
+// Distributed under the Boost Software License, Version 1.0. (See accompanying
+// file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+//
+// Official repository: https://github.com/boostorg/beast
+//
+
+// Test that header file is self-contained.
+#include <boost/beast/http/message_generator.hpp>
+
+#include <boost/beast/http/message.hpp>
+#include <boost/beast/http/string_body.hpp>
+#include <boost/beast/http/write.hpp> //for operator<<
+
+#include <boost/beast/core/buffers_to_string.hpp>
+#include <boost/beast/core/multi_buffer.hpp>
+#include <boost/beast/core/buffers_generator.hpp> // for is_buffers_generator and [async_]write overloads
+#include <boost/beast/_experimental/test/stream.hpp>
+
+#include <boost/beast/_experimental/unit_test/suite.hpp>
+
+#include <iostream>
+#include <iomanip>
+
+namespace boost {
+namespace beast {
+namespace http {
+
+namespace detail {
+
+// Degenerate body generator to trigger dynamic generator buffer
+// allocation
+//
+// Warning: magic numbers ahead
+//
+// Arbitrarily decided on 65 buffers of which two are "some" and "body", the
+// other bufs are chosen by the "seed fragment"
+struct fragmented_test_body
+{
+    using value_type = net::const_buffer;
+
+    static std::uint64_t
+    size(net::const_buffer fragment)
+    {
+        return 8 + 63 * fragment.size();
+    }
+
+    struct writer
+    {
+        using const_buffers_type = std::array<net::const_buffer, 65>;
+
+        bool done_ = false;
+        net::const_buffer seed_fragment_;
+
+        template<typename Fields>
+        writer(Fields const&, net::const_buffer fragment)
+            : seed_fragment_(fragment)
+        {
+        }
+
+        void
+        init(beast::error_code&) const
+        {
+        }
+
+        boost::optional<std::pair<const_buffers_type, bool>>
+        get(beast::error_code&)
```

Comment:
Subtle. Was there a warning about the unused `ec` arg?

Reply:
Yep, unused variable